activerecord/lib/active_record/autosave_association.rb

```diff
@@ -482,7 +507,7 @@ def save_belongs_to_association(reflection)
         return unless association && association.loaded? && !association.stale_target?
 
         record = association.load_target
-        if record && !record.destroyed?
+        if record && !record.destroyed? && !record.already_saving?
```

Comment:
Should we extract a method here? 

```ruby
def associated_record_can_save?(record)
  record && !record.destroyed? && !record.already_saving?
end
```

And then in the third case:

```ruby
next unless associated_record_can_save?(record)
```

Could also consider exposing a `record.can_save?` method from autosave_association.rb, which is `# :nodoc:` for starters.

Reply:
I think that would be a good idea.
The `can_save?` method looks cleaner to me but `associated_record_can_save?` is more inline with the rest of the code. `associated_record_can_save?` won't `pollute` the public methods either.
But maybe `can_save?` will be useful outside of autosave functionality and giving it a name now is easier than extracting it later.